### PR TITLE
added auto as themeMode prop

### PR DESCRIPTION
**Pull request by AnshumanPadiya**
#### What does this PR do?
allows auto to be passed as a value in themeMode prop - issue #6397

#### Where should the reviewer start?
Line 91 in Grommet/Grommet.js 

#### How should this be manually tested?
added themeMode story in storybook

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### Any background context you want to provide?
The changes are not

#### What are the relevant issues?
#6397 

#### Screenshots (if appropriate)
themeModeStory

authoThemeMode2

#### Do the grommet docs need to be updated?
No

### CODE REVIEW

File: src/js/components/Grommet/Grommet.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ import { format, MessageContext } from '../../contexts/MessageContext';
 import defaultMessages from '../../languages/default.json';
 import { GrommetPropTypes } from './propTypes';
 import { AnalyticsProvider } from '../../contexts/AnalyticsContext';
+// import { type } from 'os';
```
Comment (ericsoderberghp) [inline src/js/components/Grommet/Grommet.js:26]: Should remove this

File: src/js/components/Grommet/stories/ThemeMode.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+import { Grommet, Box, grommet } from 'grommet';
+
+export const ThemeMode = () => (
+  <Grommet theme={grommet} themeMode="auto">
+    <Box pad="medium">Check New Theme Mode</Box>
```
Comment (ericsoderberghp) [inline src/js/components/Grommet/stories/ThemeMode.js:7]:
```suggestion
    <Box pad="medium">"auto" themeMode</Box>
```

**Review by ericsoderberghp — COMMENTED** [review]
Nice start, just a few comments.

File: src/js/components/Grommet/propTypes.js [hunk 3]
```diff
@@ -13,7 +13,7 @@ if (process.env.NODE_ENV !== 'production') {
     plain: PropTypes.bool,
     cssVars: PropTypes.bool,
     theme: PropTypes.object,
-    themeMode: PropTypes.oneOf(['dark', 'light']),
+    themeMode: PropTypes.oneOf(['dark', 'light', 'auto']),
```
[the remark this answers is not in the document]
Comment (AnshumanPadiya) [reply src/js/components/Grommet/propTypes.js:16]: Just to be clear about it do I need to update the GrommetProps interface in the "index.d.ts" file? Please, let me know if the below update is what we need
autoThemeModeTs
Also thanks for your feedback!

[the remark this answers is not in the document]
Comment (halocline) [reply src/js/components/Grommet/propTypes.js:16]: > Please, let me know if the below update is what we need

☝️ The update you show is correct. Thank you.

**Review by jcfilben — APPROVED** [review]
Looks good! Can you create a pull request on grommet-site to add `themeMode="auto"` to the documentation?

**Comment by AnshumanPadiya** [comment]
@jcfilben Thanks, and sure thing will look into that. I had a couple of questions though if you don't mind answering, do I need to update my branch before it gets merged or will it be fine this way? also could you please suggest to me any other issue at which I could work ( it doesn't have to be included in hacktoberfest ), I would really like to make more contributions to this project.

**Comment by jcfilben** [comment]
> @jcfilben Thanks, and sure thing will look into that. I had a couple of questions though if you don't mind answering, do I need to update my branch before it gets merged or will it be fine this way? 

No need to update this branch, once it gets reviewed and approved by another person from the grommet team we will update it. If there are any merge conflicts we will let you know and ask you to fix those, but as of right now no action is needed on your end for this branch.

> also could you please suggest to me any other issue at which I could work ( it doesn't have to be included in hacktoberfest ), I would really like to make more contributions to this project.

That's awesome that you want to make more contributions to the project! Any pull requests on this repo that get approved and merged will count towards hacktoberfest even if the original issue doesn't have the hacktoberfest label. The 'hacktoberfest' label just gives people a good starting place to look for issues to work on.

As far as potential issues to work on it depends what you are interested in...
If you are interested in making Grommet more accessible see issue 6213 or filter issues by the [accessibility label](https://github.com/grommet/grommet/issues?q=is%3Aopen+is%3Aissue+label%3Aaccessibility+)

Interested in enhancements? See issue 6370 or filter issues with the [enhancement label](https://github.com/grommet/grommet/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement)

Interested in typescript? See issue 5599 or filter issues by the [typescript label](https://github.com/grommet/grommet/issues?q=is%3Aopen+is%3Aissue+label%3Atypescript)

If you come across any issues that you have questions on or are interested in working on but not sure how to approach the issue go ahead and just drop a comment on the issue & feel free to reach out with any additional questions.

**Review by ericsoderberghp — APPROVED** [review]

**Comment by jcfilben** [comment]
Awesome! Thanks for making the theme pull request

**Pull request merged**